fms/triton/telescoping_kernel.py

```diff
@@ -0,0 +1,559 @@
+import pathlib
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+import triton  # type: ignore[import-untyped]
+import triton.language as tl  # type: ignore[import-untyped]
+
+
+"""Fused telescoping cache kernel."""
+
+
+@triton.jit()
+def col_major(pid, m, n, block_m: tl.constexpr, block_n: tl.constexpr):
```

Comment:
would this be the same is swizzle2d?

also I notice block_n is unused

Reply:
not the same. column major access pattern importantly does not swizzle the tiles, which increases the perf for certain problem sizes.